pkg/pillar/diskmetrics/usage.go

```diff
@@ -195,19 +195,114 @@ func DirUsage(log *base.LogObject, dir string) (uint64, error) {
 }
 
 // Dom0DiskReservedSize returns reserved space for EVE-OS
-func Dom0DiskReservedSize(log *base.LogObject, globalConfig *types.ConfigItemValueMap, deviceDiskSize uint64) uint64 {
+// We check that the currently used aka dynamic number does not exceed
+// the statically configured percentage and number
+func Dom0DiskReservedSize(log *base.LogObject, globalConfig *types.ConfigItemValueMap, deviceDiskSize uint64, dynamicUsedByDom0 uint64) uint64 {
 	dom0MinDiskUsagePercent := globalConfig.GlobalValueInt(
 		types.Dom0MinDiskUsagePercent)
 	diskReservedForDom0 := uint64(float64(deviceDiskSize) *
 		(float64(dom0MinDiskUsagePercent) * 0.01))
-	maxDom0DiskSize := uint64(globalConfig.GlobalValueInt(
+	staticMaxDom0DiskSize := uint64(globalConfig.GlobalValueInt(
 		types.Dom0DiskUsageMaxBytes))
-	if diskReservedForDom0 > maxDom0DiskSize {
-		log.Tracef("diskSizeReservedForDom0 - diskReservedForDom0 adjusted to "+
-			"maxDom0DiskSize (%d)", maxDom0DiskSize)
-		diskReservedForDom0 = maxDom0DiskSize
+	newlogReserved := uint64(globalConfig.GlobalValueInt(types.LogRemainToSendMBytes))
+	// Always leave space for /persist/newlogd
+	maxDom0DiskSize := newlogReserved
+	// Select the larger of the current overhead usage and the configured
+	// max overhead. If using the static then ensure that we do not exceed
+	// the dom0MinDiskUsagePercent percentage of /persist
+	if staticMaxDom0DiskSize < dynamicUsedByDom0 {
+		log.Noticef("XXX Dom0DiskReservedSize using dynamic %d",
+			dynamicUsedByDom0)
+		maxDom0DiskSize += dynamicUsedByDom0
+	} else if diskReservedForDom0 > staticMaxDom0DiskSize {
+		maxDom0DiskSize += staticMaxDom0DiskSize
```

Comment:
Yes because a percentage number doesn't make much sense when you have 16TB storage.
One could argue we should always use the byte number, but I don't know if we'll see small (16GB) devices out there where we want to reserve less for dom0.

Reply:
Okay, it makes sense. By the way, I believe we can have pretty small devices, so let it be.
However, the priorities of the different values from different sources are not that obvious. It would be nice to document them and the logic behind them explicitly, at least internally, for the developers so they can answer the related questions from users quickly. In the best case it would be some user-exposed note on how to use these properties (`storage.dom0.disk.maxusagebytes`, `storage.dom0.disk.minusage.percent`)